### PR TITLE
Add new keyboard "DogTag"

### PR DESCRIPTION
**Pull request by takashicompany**
DogTag is a compact macro pad with 9 keys.  
It is assembled from a PCB no larger than 100mm x 100mm.  
It is reversible and can be used as a split macro pad by connecting two units with a TRRS cable.  
Other features include a rotary encoder and MX sockets for hot-swapping of keyswitches.  
LED underglow is also available.  

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by takashicompany** [comment]
Thank you for all your help.
I know you are busy, but please don't forget to review these Pull Requests🙏

https://github.com/qmk/qmk_firmware/pull/14949
https://github.com/qmk/qmk_firmware/pull/15077

File: keyboards/takashicompany/dogtag/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+
+    if (clockwise) {
+        tap_code(KC_MS_WH_DOWN);
+    } else {
+        tap_code(KC_MS_WH_UP);
+    }
+
+    return true;
+}
```
Comment (drashna) [inline keyboards/takashicompany/dogtag/keymaps/default/keymap.c:41]:
```suggestion
```

File: keyboards/takashicompany/dogtag/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+
+    if (clockwise) {
+        tap_code(KC_MS_WH_DOWN);
+    } else {
+        tap_code(KC_MS_WH_UP);
+    }
+
+    return true;
+}
```
Comment (drashna) [inline keyboards/takashicompany/dogtag/keymaps/via/keymap.c:41]:
```suggestion
```

File: keyboards/takashicompany/dogtag/dogtag.c [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "dogtag.h"
```
Comment (drashna) [inline keyboards/takashicompany/dogtag/dogtag.c:17]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#include "dogtag.h"

bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (clockwise) {
        tap_code(KC_MS_WH_DOWN);
    } else {
        tap_code(KC_MS_WH_UP);
    }

    return true;
}
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**